Services/communication_service.c

```diff
@@ -35,31 +53,753 @@
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_SET_AUDIO_T:
+        cnv8_16LE(&packet->data[IN_DATA_BYTE], &U_stat.set.audio_t);
+        U_stat.set.audio_t = csp_ntoh16(U_stat.set.audio_t);
+        status = HAL_UHF_setAudioT(U_stat.set.audio_t);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_SET_PARAMS:
+        cnv8_32(&packet->data[IN_DATA_BYTE], &U_stat.set.freq);
+        U_stat.set.freq = csp_ntoh32(U_stat.set.freq);
+        cnv8_16LE(&packet->data[IN_DATA_BYTE + 4],
+                &U_stat.set.PIPE_t);
+        U_stat.set.PIPE_t = csp_ntoh16(U_stat.set.PIPE_t);
+        cnv8_16LE(&packet->data[IN_DATA_BYTE + 6],
+                &U_stat.set.beacon_t);
+        U_stat.set.beacon_t = csp_ntoh16(U_stat.set.beacon_t);
+        cnv8_16LE(&packet->data[IN_DATA_BYTE + 8],
+                &U_stat.set.audio_t);
+        U_stat.set.audio_t = csp_ntoh16(U_stat.set.audio_t);
+        status =  HAL_UHF_setFreq(U_stat.set.freq)
+                + HAL_UHF_setPIPEt(U_stat.set.PIPE_t)
+                + HAL_UHF_setBeaconT(U_stat.set.beacon_t)
+                + HAL_UHF_setAudioT(U_stat.set.audio_t);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        set_packet_length(packet, sizeof(int8_t) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_RESTORE_DEFAULT:
+        status = HAL_UHF_restore(&U_restore);
+        if (sizeof(U_restore) + 1 > csp_buffer_data_size())
+        {
+            return SATR_ERROR;
+        }
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &U_restore, sizeof(U_restore));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(U_restore) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_LOW_PWR:
+        status = HAL_UHF_lowPwr(&U_stat.low_pwr_stat);
+        if (sizeof(U_stat.low_pwr_stat) + 1 > csp_buffer_data_size())
+        {
+            return SATR_ERROR;
+        }
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &U_stat.low_pwr_stat,
+               sizeof(U_stat.low_pwr_stat));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(U_stat.low_pwr_stat) + 1);
+        if (queue_response(packet) != SATR_OK)
+        {
+            return SATR_ERROR;
+        }
+        break;
+
+    case UHF_SET_DEST:
+        //uhf_struct_len = (uint8_t) packet->data[IN_DATA_BYTE]; //If using this, add +1 in the loop
+        uhf_struct_len = 6;
```

Comment:
could we change this to `sizeof(uhf_struct_t)` or whatever to avoid the magic number?